### PR TITLE
PNAPL-310: Fix broken link in readme to Rollbar API documentation

### PR DESCRIPTION
**Pull request by mangs**
**Changes:**

- Version bump to `2.0.6`
- Fix broken link in readme to Rollbar API documentation
- Update dependencies to latest

### CODE REVIEW

File: src/RollbarClientSubmitter.mts [hunk 1]
```diff
@@ -27,7 +27,9 @@ const libraryVersion = process.env['npm_package_version'];
 function buildEnumerableObject(targetObject: Error) {
   const enumerableObject: Record<string, string | Error> = {};
   for (const key of Object.getOwnPropertyNames(targetObject)) {
-    enumerableObject[key] = (targetObject as Required<Error>)[key as keyof Error];
+    const value = (targetObject as Required<Error>)[key as keyof Error];
+    enumerableObject[key] =
+      typeof value === 'string' || value instanceof Error ? value : String(value);
```
Comment (mangs) [inline src/RollbarClientSubmitter.mts:32]: The latest version of TypeScript flagged a type error here, so I fixed it

File: tsconfig.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "exactOptionalPropertyTypes": true,
     "forceConsistentCasingInFileNames": true,
     "isolatedModules": true,
-    "module": "es2022",
+    "module": "node16",
```
Comment (mangs) [inline tsconfig.json:10]: The latest version of TypeScript complained that this field did not match the value in the `moduleResolution` field, so I changed it.

File: tsconfig.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
     "skipLibCheck": true, // TODO: remove this once the underlying Vite bug is fixed: https://github.com/vitejs/vite/issues/11552
     "sourceMap": true,
     "strict": true,
-    "target": "es2021",
+    "target": "es2022",
```
Comment (mangs) [inline tsconfig.json:25]: Raising the transpile target to the max stable value

**Review by mangs — COMMENTED** [review]
Some notes for context

**Review by juanma0012 — APPROVED** [review]
LGTM!

**Pull request merged**